### DISCUSSION TITLE
Bridge Server

### DISCUSSION THREAD

**Discussion by JackLee-Zhou** [WalletConnect/walletconnect-monorepo#2758] [Developer Support]
I need to develop a bridge server to migrate my project from v1 to v2 .I have read the documentation, but the description of the protocol is not very clear. Is there any difference between v1 and v2 protocols? what should i do for this migration.Thanks !

**Comment by arein** [comment]
Hi @JackLee-Zhou, in order to migrate your project from v1 to v2 you do **not** need to develop a bridge server. Please see https://docs.walletconnect.com/2.0/advanced/migration-from-v1.x/overview for how to migrate and information on the protocol changes

**Reply by JackLee-Zhou** [reply]
but i need self-hosted WalletConnect relays,not cloud.

**Reply by arein** [reply]
Gotcha. This wasn't clear from your initial question.
May I ask why you need self-hosted?

We _currently_ do not support self-hosted v2 Relays. However, we have an entire team dedicated to decentralizing the infrastructure and enabling this again. You will hear more on that and the timeline at our announcements at ETHCC.

**Reply by tfalencar** [reply]
There are many reasons for self-hosting, including in the future, **after** v2 launched:

1- Simpler to comply with GDPR with regards to data sharing and privacy policies (e.g. what does Walletconnect and all its external dependency companies such as AWS do with customer IP's and its metadata, outside Europe?).
2- 'Federated' / self-hosting is still better than fully centralized (the case today).
3- Reduce dapp external dependencies (either in Walletconnect as a company or the future complex V2 network).
4- The "v2 fully decentralized network" still has to prove itself. Even in the event that it works, I can still see the case for dapps that **only** need a direct, 2 way communication channel, between a specific dapp and the wallet, without depending or sharing data with a broader 'V2 network'. Example:
Currently, any e-commerce owner has to host their website in **some server** (e.g., to receive orders). If there is a need for a server in any case, why not simply allow the wallet to connect to the e-commerce's website itself to establish the connection between frontend and mobile? This service doesn't even need to be "ultra scalable and ultra-low-latency" (stated goals of wc v2), since such service is only dealing with their own online store traffic, and not the whole world's walletconnect traffic.

**Reply by tfalencar** [reply]
I see the vision for wallet connect v2 as another attempt to solve similar problems that [swarm](https://www.ethswarm.org) and [waku](https://waku.org) has been working on for several years. I wish the best of luck on that front, which are problems that needs solving. As for me, I wish some open source project would exist to keep a well maintained, 'walletconnect v1-like' project.
